pkg/controllers/dashboardapi/settings/settings.go

```diff
@@ -113,11 +120,15 @@ func (s *settingsProvider) SetAll(settingsMap map[string]settings.Setting) error
 				obj.Default = setting.Default
 				update = true
 			}
-			if envValue != "" && obj.Source != "env" {
+			if envOk && obj.Source != "env" {
 				obj.Source = "env"
 				update = true
 			}
-			if envValue != "" && obj.Value != envValue {
+			if !envOk && obj.Source == "env" {
+				obj.Source = ""
+				update = true
+			}
+			if envOk && obj.Value != envValue {
```

Comment:
I think that this new code has a side effect in cases where there is a setting, with a different value and default, that is specified as `""` in the env. I don't know if this is necessarily a problem given that one of your commits is trying to explicitly support `""` as a setting value, but I did want to bring it up in case this was not intentional.

Consider the following case:
I have a setting which exists with:
`Value` = `"value"`
`Default` = `"default"`

I reboot rancher and specify this setting with an env var. The value from the env is `""`.

Under the old side code, the fallback for this would be `"value"` - since obj.Value is set to something. Under the new side code, obj.Value will be updated to `""` and fallback will be set to `"default"`, since on L135 set check to see if `obj.Value == ""` and fall back to the default case if that is true. I think that under your new approach of treating "" as a full, valid setting value, you probably want to change L135 to use `envOK` instead of `""` like you did for the other values. Although I could be wrong there, and I think that you will have a behavior change either way.

Reply:
This was actually on purpose because the transformer in the store makes it effectively this way when the setting is retrieved from the UI. The effective value of a setting should be default if the value is false.  However, I did not put enough the into the fact that this change will effect the backend. Once settings are interacted with a way that isn't settings there will also be a side effect, unless the transformer store stuff is entirely ported to webhook, which this change is trying to prevent.

Reply:
@MbolotSuse I spoke with @cbron and confirmed that this is desired. It will change the value that the backend would receive for the setting, whether that be from client or from fallback. However, the current behavior is much worse. Today someone can set `auth-user-info-max-age-seconds` using the env var to something like "600", then change it again with the env var to "", and in the API will return "" for the value (the ui will show default because of this), while the backend is still using the "600" value. We want to sync these behaviors to be the same and treat setting the value to "" properly. This means fallback being default, and Get() returning default value.